### PR TITLE
Implement BIO.tapError

### PR DESCRIPTION
**Pull request by Z1kkurat**
This resolves https://github.com/monix/monix-bio/issues/39
Probably some more tests are needed?

### CODE REVIEW

**Comment by Avasil** [comment]
Thank you @Z1kkurat :) 
I think the implementation and tests are good enough but it could use a scaladoc with a description what the function is doing. Small example would be nice as well but I'll leave it up to you

**Comment by Z1kkurat** [comment]
@Avasil I've added some docs, would be nice if you could check if that's enough

File: core/shared/src/main/scala/monix/bio/BIO.scala [hunk 1]
```diff
@@ -1827,6 +1827,27 @@ sealed abstract class BIO[+E, +A] extends Serializable {
         Map(this, f, 0)
     }
 
+  /** Creates a new task that will handle error with given function, ignoring its result
```
[the remark this answers is not in the document]
Comment (Z1kkurat) [reply core/shared/src/main/scala/monix/bio/BIO.scala]: Yes, I totally agree. Will change that line

**Comment by Z1kkurat** [comment]
Thanks for pointing out these inconsistencies @Avasil! I changed these docs in a suggested way.

**Comment by Avasil** [comment]
Awesome, I'm merging then

**Pull request merged**